multibase_test.go

```diff
@@ -68,25 +70,53 @@ func testDecode(t *testing.T, expectedEncoding Encoding, expectedBytes []byte, d
 }
 
 func TestEncode(t *testing.T) {
-	for encoding, data := range encodedSamples {
-		testEncode(t, encoding, sampleBytes, data)
+	for encoding := range EncodingToStr {
+		testEncode(t, encoding, sampleBytes, encodedSamples[encoding])
 	}
 }
 
 func TestDecode(t *testing.T) {
-	for encoding, data := range encodedSamples {
-		testDecode(t, encoding, sampleBytes, data)
+	for encoding := range EncodingToStr {
+		testDecode(t, encoding, sampleBytes, encodedSamples[encoding])
 	}
 }
 
 func TestRoundTrip(t *testing.T) {
-	buf := make([]byte, 37+16) // sufficiently large prime number of bytes (37) + another 16 to test leading 0s
+
+	for base := range EncodingToStr {
+		if int(base) == 0 {
+			// skip identity: any byte goes there
+			continue
+		}
+
+		_, _, err := Decode(string(base) + "\u00A0")
+		if err == nil {
+			t.Fatal(EncodingToStr[base] + " decode should fail on low-unicode")
+		}
+
+		_, _, err = Decode(string(base) + "\u1F4A8")
+		if err == nil {
+			t.Fatal(EncodingToStr[base] + " decode should fail on emoji")
+		}
```

Comment:
@Stebalien currently identity-base ( skipped above ) does not fail on "\u..." strings. Not sure if it should or not...

Reply:
Why would it?

Reply:
Because runes vs "bare bytes". Like... generally "low level" interfaces like this should not accept unicode in "decoded" form. But maybe in go it is idiomatic to just assume utf8 everywhere, and trip both ways correctly.

Reply:
:shrug: